x-pack/plugins/lens/public/editor_frame_service/editor_frame/config_panel/palette_indicator.tsx

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
+import { AccessorConfig } from '../../../types';
+
+export function PaletteIndicator({ accessorConfig }: { accessorConfig: AccessorConfig }) {
+  if (accessorConfig.triggerIcon !== 'colorBy' || !accessorConfig.palette) return null;
+  return (
+    <EuiFlexGroup className="lnsLayerPanel__paletteContainer" gutterSize="none" alignItems="center">
+      {accessorConfig.palette.map((color) => (
+        <EuiFlexItem
+          key={color}
+          className="lnsLayerPanel__paletteColor"
+          grow={true}
+          style={{
+            backgroundColor: color,
+          }}
+        />
+      ))}
+    </EuiFlexGroup>
```

Comment:
Rather than rolling out a custom color palette component, would it be better to use `EuiColorPaletteDisplay` instead (and override as necessary if needed)?

https://elastic.github.io/eui/#/forms/color-selection

Reply:
This component is very new - the version supporting it is not available in Kibana yet. I'm fine with using it once it becomes available, but https://github.com/elastic/kibana/pull/82730 has to get merged first.

Reply:
Sounds good. Should I open an issue as a reminder? Or is that unnecessary?

Reply:
Don't bother, I will take care of it.